File: src/com/aspect/ConnectionProvider.java
package com.aspect;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.Properties;

public class ConnectionProvider {
	static Connection con;
	static FileInputStream fis;
	static String s[]=new String[4];
	static Properties p;
	
	public static void getProperties()//Get Require Data From DBConfig.properties
	{
	try {
		fis=new FileInputStream(".//resources//DBConfig.properties");
		int x=0;
		p=new Properties();
		p.load(fis);
		
		
	s[0]=p.getProperty("DriverName");
	s[1]=p.getProperty("Url");
	s[2]=p.getProperty("UserName");
	s[3]=p.getProperty("Password");
		}catch(Exception e) {
			e.getStackTrace();
		}
	}
	
	
	public  static Connection provideConnection()  //Provide Connection
	{
		try {
			getProperties();
			Class.forName(s[0]);
			con=DriverManager.getConnection(s[1],s[2],s[3]);
			System.out.println("Connection Established");
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
		} catch (SQLException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	return con;
	
	}
	
}

File: src/com/aspect/ConnectionProvider.go
package aspect

import (
	"database/sql"
	"fmt"
	"io/ioutil"
	"log"
	"strings"

	_ "github.com/go-sql-driver/mysql"
)

func GetProperties() string {
	data, err := ioutil.ReadFile("../../resources/DBConfig.properties")
	if err != nil {
		fmt.Println("File reading error", err)
	}
	//fmt.Println("Contents of file:")
	//fmt.Println(string(data))
	splitedData := strings.Split(string(data), "\n")
	//fmt.Println(splitedData)
	Url := strings.Split(splitedData[1], "=")[1]
	UserName := strings.Split(splitedData[2], "=")[1]
	Password := strings.Split(splitedData[3], "=")[1]
	Database := strings.Split(splitedData[4], "=")[1]
	return fmt.Sprintf("%v:%v@tcp(%v)/%v", UserName, Password, Url, Database)
}

func ProvideConnection() *sql.DB {

	db, err := sql.Open("mysql", GetProperties())

	if err != nil {
		log.Fatal(err)
	}
//	defer db.Close()
	//Call db.Ping() to check the connection
	pingErr := db.Ping()
	if pingErr != nil {
		log.Fatal(pingErr)
	}
	fmt.Println("Connected!")
	return db
}
